big_tests/tests/inbox_extensions_SUITE.erl

```diff
@@ -0,0 +1,725 @@
+                       muclight_domain/0,
+                       required_modules/0,
+                       inbox_opts/0
+                      ]).
+
+
+all() ->
+    inbox_helper:skip_or_run_inbox_tests([{group, inbox_extensions}]).
+
+suite() ->
+    escalus:suite().
+
+groups() ->
+    G = [{inbox_extensions, [], inbox_extensions_tests()}],
+    ct_helper:repeat_all_until_all_ok(G).
+
+inbox_extensions_tests() ->
+    [
+     {generic, [parallel], [
+        % General errors
+        returns_error_when_no_jid_provided,
+        returns_error_when_invalid_jid_provided,
+        returns_error_when_valid_jid_but_no_property,
+        % Set-unread errors
+        returns_error_when_read_invalid_value,
+        returns_error_when_read_valid_request_but_not_in_inbox,
+        % Archiving errors
+        returns_error_when_archive_invalid_value,
+        returns_error_when_archive_valid_request_but_not_in_inbox,
+        % Muting errors
+        returns_error_when_mute_invalid_value,
+        returns_error_when_mute_invalid_seconds,
+        returns_error_when_mute_valid_request_but_not_in_inbox,
+        % Form errors
+        returns_error_when_archive_field_is_invalid,
+        returns_error_when_max_is_not_a_number
+      ]},
+     {one_to_one, [parallel], [
+        % read
+        read_unread_entry_set_to_read,
+        read_read_entry_set_to_unread,
+        read_unread_entry_with_two_messages_when_set_unread_then_unread_count_stays_in_two,
+        % archive
+        archive_active_entry_gets_archived,
+        archive_archived_entry_gets_active_on_request,
+        archive_archived_entry_gets_active_for_the_sender_on_new_message,
+        archive_archived_entry_gets_active_for_the_receiver_on_new_message,
+        archive_active_unread_entry_gets_archived_and_still_unread,
+        archive_full_archive_can_be_fetched,
+        % mute
+        mute_unmuted_entry_gets_muted,
+        mute_muted_entry_gets_unmuted,
+        mute_after_timestamp_gets_unmuted,
+        mute_muted_conv_restarts_timestamp,
+        % other
+        returns_valid_properties_form,
+        properties_can_be_get,
+        properties_many_can_be_set,
+        max_queries_can_be_limited,
+        max_queries_can_fetch_ahead,
+        timestamp_is_not_reset_with_setting_properties
+      ]},
+     {muclight, [sequence], [
+        groupchat_setunread_stanza_sets_inbox
+      ]}
+    ].
+
+init_per_suite(Config) ->
+    ok = dynamic_modules:ensure_modules(inbox_helper:domain(), required_modules()),
+    InboxOptions = inbox_opts(),
+    Config1 = escalus:init_per_suite(Config),
+    Config2 = [{inbox_opts, InboxOptions} | Config1],
+    escalus:create_users(Config2, escalus:get_users([alice, bob, kate, mike])).
+
+end_per_suite(Config) ->
+    Host = ct:get_config({hosts, mim, domain}),
+    Config1 = escalus:delete_users(Config, escalus:get_users([alice, bob, kate, mike])),
+    dynamic_modules:stop(Host, mod_inbox),
+    dynamic_modules:stop(Host, mod_muc_light),
```

Comment:
Let's refactor this with `save_modules` and `restore_modules` in the future, together with `inbox_SUITE`.